profiles/views.py

```diff
@@ -37,6 +43,17 @@ class ProfileViewSet(RetrieveModelMixin, UpdateModelMixin, GenericViewSet):
     serializer_class_filled_out = ProfileFilledOutSerializer
     serializer_class_limited = ProfileLimitedSerializer
 
+    def perform_update(self, serializer):
+        """Updates object"""
+        profile = serializer.save()
+        if 'email_optin' in serializer.validated_data:
+            # perform mailgun action if email optin is set then remove user from mailgun unsubscription
+            # list otherwise add user to mailgun unsubscription.
```

Comment:
This comment can be removed when you make the changes that I recommended above. It will be obvious what the code is doing